cscs-checks/libraries/io/hdf5_compile_run.py

```diff
@@ -19,10 +19,16 @@ def __init__(self, lang, linkage):
         self.descr = lang_names[lang] + ' HDF5 ' + linkage.capitalize()
         self.sourcepath = f'h5ex_d_chunk.{lang}'
         self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc']
-        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
-                                    'PrgEnv-intel', 'PrgEnv-pgi']
         if linkage == 'dynamic':
-            self.valid_systems += ['eiger:mc']
+            self.valid_systems += ['eiger:mc', 'pilatus:mc']
+
+        # PrgEnv-intel on Pilatus does not feature cray-hdf5 as of PE 21.02
+        if self.current_system.name == 'pilatus':
```

Comment:
and eiger ? and PrgEnv-intel ?

Reply:
The exception is only for Pilatus since at present `PrgEnv-intel` does not feature `cray-hdf5` in PE 21.02. 
There is no `PrgEnv-intel` on Eiger at the moment, so this is not defined in the `config.py` for the system.

Reply:
You can check yourself that the `intel` compiler is missing in the output of the command `module spider cray-hdf5/1.12.0.3`: 
```
uan04:~$ module spider cray-hdf5/1.12.0.3
----------------------------------------------------------------------------------------------------------------------------------
  cray-hdf5: cray-hdf5/1.12.0.3
----------------------------------------------------------------------------------------------------------------------------------
    You will need to load all module(s) on any one of the lines below before the "cray-hdf5/1.12.0.3" module is available to load.
      aocc/2.2.0.1
      cce/10.0.4
      cce/11.0.0
      cce/11.0.1
      cce/11.0.2
      cce/11.0.3
      gcc/10.1.0
      gcc/10.2.0
      gcc/9.3.0
```